WordPress/Classes/ViewRelated/Jetpack/Jetpack Restore/JetpackRestoreViewController.swift

```diff
@@ -0,0 +1,169 @@
+
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureTitle()
+        configureNavigation()
+        configureTableView()
+        reloadViewModel()
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        tableView.layoutHeaderView()
+    }
+
+    // MARK: - Configure
+
+    private func configureTitle() {
+        switch restoreAction {
+        case .restore:
+            title = NSLocalizedString("Restore", comment: "Title for the Jetpack Restore Site Screen")
+        case .downloadBackup:
+            title = NSLocalizedString("Download Backup", comment: "Title for the Jetpack Download Backup Site Screen")
+        }
+    }
+
+    private func configureNavigation() {
+        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel,
+                                                           target: self,
+                                                           action: #selector(cancelTapped))
+    }
+
+    private func configureTableView() {
+        WPStyleGuide.configureColors(view: view, tableView: tableView)
+
+        ImmuTable.registerRows([SwitchRow.self], tableView: tableView)
+
+        let headerView = JetpackRestoreHeaderView.loadFromNib()
+        headerView.configure(site: site, formattableActivity: activity, restoreAction: restoreAction)
+        self.tableView.tableHeaderView = headerView
+    }
+
+    // MARK: - Private Helpers
+
+    @objc private func cancelTapped() {
+        self.dismiss(animated: true)
+    }
+
+    // MARK: - Model
+
+    private func reloadViewModel() {
+        handler.viewModel = tableViewModel()
+    }
+
+    private func tableViewModel() -> ImmuTable {
+        return ImmuTable(
+            sections: [
+                generalSection(),
+                contentSection(),
+                databaseSection()
+            ]
+        )
+    }
+
+    private func generalSection() -> ImmuTableSection {
+        let themesRow = SwitchRow(
+            title: NSLocalizedString("WordPress Themes", comment: "Downloadable/Restorable items: WordPress Themes"),
+            value: true,
+            onChange: { _ in }
+        )
+        let pluginsRow = SwitchRow(
+            title: NSLocalizedString("WordPress Plugins", comment: "Downloadable/Restorable items: WordPress Plugins"),
+            value: true,
+            onChange: { _ in }
+        )
+        let mediaUploadsRow = SwitchRow(
+            title: NSLocalizedString("Media Uploads", comment: "Downloadable/Restorable items: Media Uploads"),
+            value: true,
+            onChange: { _ in }
+        )
+        let rootRow = SwitchRow(
+            title: NSLocalizedString("WordPress root", comment: "Downloadable/Restorable items: WordPress root"),
+            value: true,
+            onChange: { _ in }
+        )
+
+        let headerText: String
+        switch restoreAction {
+        case .restore:
```

Comment:
We already have this `switch` statement in a few different parts. This makes me wonder if we shouldn't break this VC (either by using inheritance or composition).

Right now, as it is, is not that complex. But I fear that this might grow into something huge and messy in the future.

We could have something like `JetpackRestoreViewController` and `JetpackBackupViewController`.

What do you think?

(An example of usage of inheritance (although it has it's own problems too tbh) is the `ReaderCardsStreamViewController`)

Reply:
@leandroalonso 

> Right now, as it is, is not that complex. But I fear that this might grow into something huge and messy in the future.

That's a good point. I'll address this in a future PR, maybe via protocols!